app/src/main/java/org/simple/clinic/contactpatient/ContactPatientUpdate.kt

```diff
@@ -28,9 +30,24 @@ class ContactPatientUpdate(
       is PatientAgreedToVisitClicked -> dispatch(MarkPatientAsAgreedToVisit(model.appointment!!.get().appointment.uuid))
       is NextReminderDateClicked -> selectNextReminderDate(model)
       is PreviousReminderDateClicked -> selectPreviousReminderDate(model)
+      is ManualDateSelected -> updateWithManuallySelectedDate(event, model)
     }
   }
 
+  private fun updateWithManuallySelectedDate(
+      event: ManualDateSelected,
+      model: ContactPatientModel
+  ): Next<ContactPatientModel, ContactPatientEffect> {
+    val selectedDate = event.selectedDate
+    val currentDate = event.currentDate
+
+    val matchingPotentialAppointmentDate = model.potentialAppointments.firstOrNull { it.scheduledFor == selectedDate }
+
+    val reminderDate = matchingPotentialAppointmentDate ?: PotentialAppointmentDate(selectedDate, Days(currentDate daysTill selectedDate))
```

Comment:
Selected date can't be less than current date right?

Reply:
It cannot. This is being controlled on the UI since the date pickers are disabled when the selected date is a day after the current date.